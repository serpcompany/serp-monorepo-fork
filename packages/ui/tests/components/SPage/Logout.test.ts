import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it, vi } from 'vitest';
import { ref } from 'vue';
import Logout from '../../../components/SPage/Logout.vue';
import ComponentRender from '../../componentRender';

const userSessionMock = {
  loggedIn: ref(true),
  user: ref({ name: 'Test User', email: 'test@test.com' }),
  clear: vi.fn()
} as unknown;

mockNuxtImport('useUserSession', () => () => userSessionMock);

describe('SPageLogout Snapshot', () => {
  const scenarios: [string, { loggedIn: boolean }][] = [
    ['when user is logged in', { loggedIn: true }],
    ['when user is not logged in', { loggedIn: false }]
  ];

  it.each(scenarios)(
    'handles %s correctly',
    async (desc: string, { loggedIn }) => {
      // Update the mock state for this test scenario.
      userSessionMock.loggedIn.value = loggedIn;

      // Mock navigateTo to capture redirection calls.
      mockNuxtImport('navigateTo', () => vi.fn());

      // Render the Logout component.
      const html = await ComponentRender(`Logout ${desc}`, {}, Logout);
      expect(html).toMatchSnapshot();
    }
  );
});
