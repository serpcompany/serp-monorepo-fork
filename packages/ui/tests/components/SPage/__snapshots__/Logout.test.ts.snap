// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPageLogout Snapshot > handles when user is logged in correctly 1`] = `
"<div><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-sm gap-1.5 text-inverted bg-primary hover:bg-primary/75 disabled:bg-primary aria-disabled:bg-primary focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary">
    <!--v-if--> Sign Out
    <!--v-if-->
  </button></div>"
`;

exports[`SPageLogout Snapshot > handles when user is not logged in correctly 1`] = `
"<div><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-sm gap-1.5 text-inverted bg-primary hover:bg-primary/75 disabled:bg-primary aria-disabled:bg-primary focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary">
    <!--v-if--> Sign Out
    <!--v-if-->
  </button></div>"
`;
